staging/src/k8s.io/component-base/featuregate/feature_gate.go

```diff
@@ -296,6 +307,29 @@ func (f *featureGate) Add(features map[Feature]FeatureSpec) error {
 	return nil
 }
 
+func (f *featureGate) OverrideDefault(name Feature, override bool) error {
+	f.lock.Lock()
+	defer f.lock.Unlock()
+
```

Comment:
I think that we should return an error when this `featureGate` has already been registered with a `FlagSet` (call to `AddFlag` method).

Reply:
This should work. For example: KCM options are created first and only then are exposed as command line options (https://github.com/kubernetes/kubernetes/blob/3edcce52e3ce8a61bf7ef8fa2becd8fa0fa0d0ed/cmd/kube-controller-manager/app/controllermanager.go#L157)